src/controllers/controller_post.rs

```diff
@@ -27,11 +27,18 @@ pub async fn show_post(Path(post_id_tmp): Path<i32>) -> impl IntoResponse {
         category_name = index.category_name.clone();
     }
     let category_list = get_all_categories().await;
-    category_list.iter().for_each(|categories| {
-        categories.iter().for_each(|category| {
-            category_in_template.push(category.clone().category_name);
-        })
-    });// todo add flatten() here
+    // category_list.iter().for_each(|categories| {
+    //     categories.iter().for_each(|category| {
+    //         category_in_template.push(category.clone().category_name);
+    //     })
+    // });//
+    let tmp = category_list.into_iter()
+        .map(|x| x.into_iter().collect::<Vec<_>>())
+        .flatten()
+        .collect::<Vec<_>>();
+    tmp.iter().for_each(|category| {
+        category_in_template.push(category.clone().category_name);
+    });
```

Comment:
I think the flatten is redundant here.
`category_list` itself is a type of `Result<Vec<Category>>`. All you need to do is simply propagate the error or using `unwrap` to get the inner value

```rust
    //let category_list = get_all_categories().await;
    //let tmp = category_list.into_iter()
    //    .map(|x| x.into_iter().collect::<Vec<_>>())
    //    .flatten()
    //    .collect::<Vec<_>>();
    let category_in_template = 
        get_all_categories()
          .await
          .unwrap() // or using `?` to handle error case
          .into_iter()
          .map(|category| category.category_name)
          .collect::<Vec<_>>();
```

Reply:
It's fixed